Code/Framework/AzCore/AzCore/Asset/AssetManager.cpp

```diff
@@ -27,6 +27,13 @@
 #include <utility>
 #include <AzCore/Serialization/ObjectStream.h>
 
+#define ENABLE_DEBUG_OUTPUT 0
```

Comment:
Putting this behind a compile flag like this makes me nervous, we have others, like the logging for SQL commands, that sometimes break because we never check them

Reply:
Fair point, I wanted to do it with a constexpr if but the compiler didn't seem to be optimizing away the parameter function calls, which meant it always had an overhead.  This is a pretty core system so I think we should be careful about adding unnecessary overhead.  Do you feel strongly about changing this to a `constexpr if` or have another solution?

Reply:
I guess we can start with the #defines for now and try to come up with something better, later. This is such a niche feature, I think most people who will need to look into it could get it re-compiling if it breaks, you're going to have to be pretty expert at the asset pipeline to get to the point of knowing this exists, and wanting to make use of it.